IFC 4.3/LinearPlacement_1/UT_LinearPlacement_1.ifc

```diff
@@ -1,3 +1,113 @@
+#25=IFCALIGNMENT2DHORIZONTALSEGMENT($,$,$,#23);
+
+#26=IFCCIRCULARARCSEGMENT2D(#27,1.5707963267948966,785.39816339744823,500.,.F.);
+#27=IFCCARTESIANPOINT((0.,1000.));
+#28=IFCALIGNMENT2DHORIZONTALSEGMENT($,$,$,#26);
+
+/* Vertical segments */
+
+
+/* --- --- --- --- --- --- --- --- */
+
+/* -- Geometry of product*/
+/* Note: The placement point is NOT the center of mass! */
+#50=IFCCARTESIANPOINTLIST3D(((10.,20.,50.),(0.,20.,50.),(10.,0.,50.),(10.,0.,50.),(0.,20.,50.),(0.,0.,50.),(10.,0.,0.),(0.,0.,0.),(10.,20.,0.),(10.,20.,0.),(0.,0.,0.),(0.,20.,0.),(0.,0.,50.),(0.,0.,0.),(10.,0.,50.),(10.,0.,50.),(0.,0.,0.),(10.,0.,0.),(0.,20.,50.),(0.,20.,0.),(0.,0.,50.),(0.,0.,50.),(0.,20.,0.),(0.,0.,0.),(10.,20.,50.),(10.,20.,0.),(0.,20.,50.),(0.,20.,50.),(10.,20.,0),(0,20.,0),(10.,0.,50.),(10.,0.,0.),(10.,20.,50.),(10.,20.,50.),(10.,0.,0.),(10.,20.,0.)),$);
+#51=IFCTRIANGULATEDFACESET(#50,$,.T.,((1,2,3),(4,5,6),(7,8,9),(10,11,12),(13,14,15),(16,17,18),(19,20,21),(22,23,24),(25,26,27),(28,29,30),(31,32,33),(34,35,36)),$);
+#52=IFCSHAPEREPRESENTATION(#15,'Body','Tessellation',(#51));
+#53=IFCPRODUCTDEFINITIONSHAPE($,$,(#52));
+
+/* -- IfcProduct instance with IfcLinearPlacement */
+#90=IFCBUILDINGELEMENTPROXY('2uEmTbIVTEr94hFPqz1_N9',#2,'linear positioned in arc segment',$,$,#91,#53,$,$);
+/* -- IfcLinearPlacement: station 600m, 14m lateral offset, 0m vertical offset, 0m logitudinal offset, measurement should be applied the horizontal projection (has no impact in this sample since no vertical alignment has been defined) */
+#91=IFCLINEARPLACEMENT($,#20,#92,#93,$);
+#92=IFCDISTANCEEXPRESSION(600.,-14.,0.,0.,.T.);
+#93=IFCORIENTATIONEXPRESSION(#94,#95);
+#94=IFCDIRECTION((1.,0.,0.));
+#95=IFCDIRECTION((0.,0.,1.));
+
+/* -- IfcProduct instance with IfcLinearPlacement, placement at arc segment */
+#100=IFCBUILDINGELEMENTPROXY('2uEmTbIVTEr94hFPqz1_M9',#2,'linear positioned',$,$,#101,#53,$,$);
+
+/* -- IfcLinearPlacement for #100:  */
+#101=IFCLINEARPLACEMENT($,#20,#102,#103,$);
+#102=IFCDISTANCEEXPRESSION(1500.,-14.,0.,0.,.T.);
+#103=IFCORIENTATIONEXPRESSION(#104,#105);
+#104=IFCDIRECTION((0.,0.,1.));
+#105=IFCDIRECTION((1.,0.,0.));
+
+/* -- IfcProduct instance with IfcLocalPlacement - no rotation */
+#110=IFCBUILDINGELEMENTPROXY('3$fw5brl95uh4EJX8NGPg7',#2,'euclidian positioned','localPos',$,#111,#53,$,$);
+
+/* -- IfcProduct instance with local/euclidian placement */
+#111=IFCLOCALPLACEMENT($,#112);
+#112=IFCAXIS2PLACEMENT3D(#113,#114,#115);
+#113=IFCCARTESIANPOINT((100.,0.,0.));
+#114=IFCDIRECTION((1.,0.,0.));
+#115=IFCDIRECTION((0.,0.,1.));
+
+/* -- IfcProduct instance with IfcLocalPlacement - rotated */
+#120=IFCBUILDINGELEMENTPROXY('3$fw5brl95uh4EJX8NGPf7',#2,'euclidian and rotated positioned','localPos + Rotation',$,#121,#53,$,$);
+
+/* -- IfcProduct instance with local/euclidian placement - rotated */
+#121=IFCLOCALPLACEMENT($,#122);
+#122=IFCAXIS2PLACEMENT3D(#123,#124,#125);
+#123=IFCCARTESIANPOINT((100.,20.,0.));
+#124=IFCDIRECTION((.4142,.4142,0.));
```

Comment:
This is not valid, the token for REAL is:

> REAL              = [ SIGN ] DIGIT { DIGIT } "." { DIGIT }
>                            [ "E" [ SIGN ] DIGIT { DIGIT } ] .


So you have to have a digit before the point. Annoying because most modern languages do allow to omit the leading 0.

Reply:
Thank you for pointing it out. See commit 6837c62